### PR TITLE
CMS - set auth option back to default when cancelled

### PR DESCRIPTION
**Pull request by abist**
Edit - had the wrong issue. Here's the correct one - https://github.com/microsoft/azuredatastudio/issues/5856

### CODE REVIEW

File: src/sql/workbench/services/connection/browser/cmsConnectionWidget.ts [hunk 1]
```diff
@@ -108,8 +108,9 @@ export class CmsConnectionWidget extends ConnectionWidget {
 			} else {
 				authTypeOption.defaultValue = this.getAuthTypeDisplayName(AuthenticationType.SqlLogin);
 			}
-			this._authTypeSelectBox.setOptions(authTypeOption.categoryValues.map(c => c.displayName), 1);
 		}
+		this._authTypeSelectBox.setOptions(authTypeOption.categoryValues.map(c => c.displayName), 1);
+		this._authTypeSelectBox.selectWithOptionName(authTypeOption.defaultValue);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/connection/browser/cmsConnectionWidget.ts]: Does this work as expected with registered servers? You're calling setOptions twice in that case.

File: src/sql/workbench/services/connection/browser/cmsConnectionWidget.ts [hunk 2]
```diff
@@ -108,8 +108,9 @@ export class CmsConnectionWidget extends ConnectionWidget {
...
 	}
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/connection/browser/cmsConnectionWidget.ts]: How does this fix the issue described? An error when expanding the node doesn't seem like it should be affected by the default value in the connection dialog...

File: src/sql/workbench/services/connection/browser/cmsConnectionWidget.ts (hunk 1 above)
Comment (abist) [reply src/sql/workbench/services/connection/browser/cmsConnectionWidget.ts]: Yes, it defaults to the default value when cancelled.

File: src/sql/workbench/services/connection/browser/cmsConnectionWidget.ts (hunk 2 above)
Comment (abist) [reply src/sql/workbench/services/connection/browser/cmsConnectionWidget.ts]: Default values matter when setting up the authorization type

Comment (Charles-Gagnon) [reply src/sql/workbench/services/connection/browser/cmsConnectionWidget.ts]: But how is that impacting the connection that is made when you expand a node in the tree? That doesn't necessarily involve opening the connection dialog at all.

Comment (abist) [reply src/sql/workbench/services/connection/browser/cmsConnectionWidget.ts]: Just realized I had the incorrect issue tagged here. Updated the issue. Sorry for the confusion!

**Comment by abist** [comment]
Closing this as it's going be overwritten by https://github.com/microsoft/azuredatastudio/pull/5873

**Pull request closed without merging**